timmy/nodes.py

```diff
@@ -511,14 +517,14 @@ def nodes_init(self):
                 self.nodes[node.ip] = node
 
     def nodes_get_release(self):
-        run_items = []
+        known_clusters = {}
         for key, node in self.nodes.items():
-            if not node.filtered_out:
-                run_items.append(tools.RunItem(target=node.get_release,
-                                               key=key))
-        result = tools.run_batch(run_items, 100, dict_result=True)
-        for key in result:
-            self.nodes[key].release = result[key]
+            if not node.filtered_out and not node.release:
+                if known_clusters.get(node.cluster):
```

Comment:
We usually use 'x not in y', i.e. 'node.cluster not in known_clusters' - more readable. Same goes for the next line.